### ISSUE TITLE
Mono: The type initializer for 'Un4seen.Bass.Bass' threw an exception.

### ISSUE THREAD

**Issue by konsumer** [fstarred/xrns2xmod#12]
I am trying to use it on linux.

I can run 

```
mono Xrns2XModShell.exe
```

and get the CLI help, so I try to run it on a 3.2.2 renoise file, and get htis:

```
mono Xrns2XModShell.exe simplemod.xrns --type=mod --out=simple.mod
The type initializer for 'Un4seen.Bass.Bass' threw an exception.
```

Here is my test file:
[simplemod.zip](https://github.com/fstarred/xrns2xmod/files/6133161/simplemod.zip)

**Comment by deadpahn** [comment]
Yeah I got this too. I tried grabbing an old version, 3, and running it in steam using proton 7. The old version comes with a ui and it actually worked (except that I was trying to load a new file so the compatibility check broke it)

The error isn't verbose enough to tell us what is happening.

For fun, I think running renoise under proton for this might work. Maybe if the UI was still there (perhaps it's there in the build somewhere). I dunno I'd have to dig a bit more but at this point I dont have time for that.

Sad to see this issue is still open after 2 years.

**Comment by konsumer** [comment]
Maybe we need to just make a nice cross-platform version...

**Comment by deadpahn** [comment]
It is cross platform. The error is with one of its dependencies, Un4seen.

https://www.un4seen.com/

It's throwing errors on our distros, not sure why I didn't have time to
look at it.

**Comment by konsumer** [comment]
Well, I mean using a library that is cross-platform and works on more systems easily, like openmpt for messing with mod/xm or ffmpeg for converting things.  I am pretty busy right now, but maybe I will take a stab at it soon.

**Comment by deadpahn** [comment]
Interesting. I had a look and I'm not sure if it's something I'll tackle.

**Comment by prayerie** [comment]
also having this issue on macos

**Comment by henrikth93** [comment]
Hello! I am trying to look into this - is there a channel we can use for discussions, like discord?

**Comment by lutalli** [comment]
I was experiencing the same issue and I found a solution (Arch Linux):

Get

- `libbass.so`
- `libbassflac.so`
- `libbassmix.so`

and place them in the the same directory where `Xrns2XModShell.exe` is located. Then the program should work.

The first one is the BASS audio library, the other two are extensions.

To get these `.so` files, go to [Un4seen](https://www.un4seen.com) > BASS and download the corresponding files for Linux. 

For example, to download the library, click on the "Linux" button at top of the page and you'll get `bass24-linux.zip`. After extraction `libbass.so` can be found in `bass24-linux/libs/<platform>` (`x86_64` for me).

I can't explain exactly how this is happening but anyway this fixed it.

The Renoise menu tool also works now with this fix.

**Comment by konsumer** [comment]
This is great advice for others. It was a long time ago, but if I remember right, I tried, and could not get it to work like this on PopOS, at the time. I ended up making [renoise2mod](https://github.com/konsumer/renoise2mod), which is very similar, but doesn't use proprietary audio-lib, and is written in rust (CLI, with renoise tool UI) for maximum cross-compat. It could use more testing, if you guys want to try it out.

I am going to close this, for now, as it seems like you got it resolved, and I don't need it anymore.

**Comment by lutalli** [comment]
> I ended up making [renoise2mod](https://github.com/konsumer/renoise2mod)

That's some cool stuff! Concerning how this project is in such a dependency mess for Linux and, as far as I see it, not truly cross-platform, I did look around to see if there is any better rewrites, but I didn't find any.

**Comment by konsumer** [comment]
I published [here](https://www.renoise.com/tools/renoise2mod)
